tests/pch/tests-pch.hpp

```diff
@@ -1,3 +1,3 @@
 #include "../pch/main-pch.hpp"
 #define CATCH_CONFIG_ALL_PARTS
-#include "catch/catch.hpp"
+#include "../catch/catch.hpp"
```

Comment:
I'm not sure why this is needed now, but not before.

Reply:
I don't think it's using the PCH at all. You're outputting the PCH to [`tests-pch-tiles-sound-back.hpp.gch`](https://github.com/CleverRaven/Cataclysm-DDA/runs/1711048834?check_suite_focus=true#step:8:660), but still only including [`tests-pch.hpp`](https://github.com/CleverRaven/Cataclysm-DDA/runs/1711048834?check_suite_focus=true#step:8:663). The header and its PCH need to have the [same name](https://gcc.gnu.org/onlinedocs/gcc/Precompiled-Headers.html), which is another reason why using a build prefix/dir is much better IMO.

The rule for the PCH uses `-I.` unlike the rule for test objects. Since you're getting that inclusion error, it means that gcc is not using the precompiled header.

Reply:
That's 100% correct, and I did see it before.
Now, I'm not so sure, but I think, I also see it include suffixed PCH, maybe by clang. I'll try to look it up.

Reply:
I think, I've finally understood, your suggestion. Targets should be separate for PCH builds from others, is that correct? Or am I still missing it?